elastic/kibana
# [Infra UI] Fix theme passed in to charts

closes [#110217](https://github.com/elastic/kibana/issues/110217)
## Summary

This PR addresses an issue with how the infra sets the theme in charts. It replaces the previous implementation with the use of `useChartsBaseTheme` and `useChartsTheme` provided by the charts service, ensuring consistent behavior for both light and dark modes. From a UI perspective, this change doesn't introduce any alterations, but it solves the problem related to dark mode.

### Screenshots

#### Node Details
<img width="899" alt="image" src="https://github.com/elastic/kibana/assets/2767137/7ec8b9e9-c440-44d1-86ac-45dbab76a8a9">

<img width="899" alt="image" src="https://github.com/elastic/kibana/assets/2767137/c6aac4a6-8982-47cc-b08e-ee5d8af0391f">


#### Infra
##### Timeline
<img width="720" alt="image" src="https://github.com/elastic/kibana/assets/2767137/cd4a1a3c-b5a4-47d5-bbde-5bf54dfac262">

<img width="720" alt="image" src="https://github.com/elastic/kibana/assets/2767137/839ccaf0-02a5-401e-957a-137d0ef25605">


##### Metrics
<img width="541" alt="image" src="https://github.com/elastic/kibana/assets/2767137/69138a89-21ff-4f17-bcbd-64ac609dd9e7">

<img width="541" alt="image" src="https://github.com/elastic/kibana/assets/2767137/63b02a7a-397e-451e-b4d7-7b37fdc4f6a9">


##### Processes
<img width="541" alt="image" src="https://github.com/elastic/kibana/assets/2767137/01e7d050-0103-4de6-8215-e93b9f80751b">

<img width="541" alt="image" src="https://github.com/elastic/kibana/assets/2767137/bd1ce95a-1944-411f-abc4-8f2a998f7643">


### Metrics Explorer
<img width="720" alt="image" src="https://github.com/elastic/kibana/assets/2767137/60d2857a-3b97-4fd6-8cc7-8b7ab757b3d5">

<img width="720" alt="image" src="https://github.com/elastic/kibana/assets/2767137/98bba0ea-27db-48a2-ae20-ff4162212c6e">


### Alert
#### Inventory rule
<img width="450" alt="image" src="https://github.com/elastic/kibana/assets/2767137/8b787e53-cb47-4254-ab69-69ab22ff0388">

<img width="450" alt="image" src="https://github.com/elastic/kibana/assets/2767137/a952bf6b-e22f-4753-8db6-f2deb213b22a">


#### Metric Threshold rule
<img width="450" alt="image" src="https://github.com/elastic/kibana/assets/2767137/db3d1448-5faa-4e20-bda8-7351d9b52144">

<img width="450" alt="image" src="https://github.com/elastic/kibana/assets/2767137/a11bc2de-f042-4d4f-9adc-d376205bc23f">


#### Logs rule
<img width="450" alt="image" src="https://github.com/elastic/kibana/assets/2767137/b703e972-2d47-485e-a2f5-4cfca0c982cd">

<img width="450" alt="image" src="https://github.com/elastic/kibana/assets/2767137/a98d3827-d930-4ef2-90f9-f336583f15da">




### How to tests
- Set up a local Kibana instance.
- Navigate to the pages mentioned above.
  - Change the user settings to both dark and light mode.
      - In light mode, ensure that there are no differences in the charts when compared to the edge cluster.
      - In dark mode, confirm that the issue described in the ticket is resolved (no horizontal line in white).
